docusaurus/docs/tutorials/convert-a-frontend-datasource-to-backend.md

```diff
@@ -0,0 +1,415 @@
+
+### Health check
+
+Once you move authentication logic to the backend, you can do a health check in the backend.
+
+:::note
+
+You need to delete the frontend implementation `testDatasource` in your `Datasource` class in the frontend to use the health check in the backend.
+
+:::
+
+In this frontend example, the health check makes an API request to `https://api.example.com` (as defined in the `routes` field in `plugin.json`) and returns an error if the request fails:
+
+```typescript title="src/DataSource.ts"
+import { getBackendSrv } from '@grafana/runtime';
+
+const routePath = '/example';
+
+export class MyDatasource extends DataSourceApi<MyQuery, MyDataSourceJsonData> {
+  ...
+
+  async testDatasource() {
+    try {
+      await getBackendSrv().datasourceRequest({
+        url: this.url + routePath + '/v1/users',
+        method: 'GET',
+      });
+      return {
+        status: 'success',
+        message: 'Health check passed.',
+      };
+    } catch (error) {
+      return { status: 'error', message: error.message };
+    }
+  }
+}
+```
+
+In the case of a backend data source, the `Datasource` struct should implement the `CheckHealth` method. This method returns an error if the data source is not healthy. For example:
+
+```go title="pkg/plugin/datasource.go"
+func NewDatasource(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	opts, err := settings.HTTPClientOptions(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("http client options: %w", err)
+	}
+
+	cl, err := httpclient.New(opts)
+	if err != nil {
+		return nil, fmt.Errorf("httpclient new: %w", err)
+	}
+
+return &Datasource{
+		settings:   settings,
+		httpClient: cl,
+	}, nil
+}
+
+func (d *Datasource) CheckHealth(ctx context.Context, _ *backend.CheckHealthRequest) (*backend.CheckHealthResult, error) {
+	resp, err := d.httpClient.Get(d.settings.URL + "/v1/users")
+	if err != nil {
+    // Log the error here
+  	return &backend.CheckHealthResult{
+	  	Status: backend.HealthStatusError,
+		  Message: "request error",
+	  }, nil
+	}
+	if resp.StatusCode != http.StatusOK {
+  	return &backend.CheckHealthResult{
+	  	Status: backend.HealthStatusError,
+		  Message: fmt.Sprintf("got response code %d", resp.StatusCode),
+	  }, nil
+	}
+	return &backend.CheckHealthResult{
+		Status:  backend.HealthStatusOk,
+		Message: "Data source is working",
+	}, nil
+}
+```
+
+:::note
+
+This example covers an HTTP-only data source. So, if your data source requires a database connection, you can use the Golang client for the database and execute a simple query like `SELECT 1` or a `ping` function.
+
+:::
+
+### Query
+
+The next step is to move the query logic. This will highly vary depending on how the plugin queries the data source and transforms the response into [frames](../introduction/data-frames.md). In this tutorial, let's see how we can migrate a simple example.
```

Comment:
```suggestion
The next step is to move the query logic. This will significantly vary depending on how the plugin queries the data source and transforms the response into [frames](../introduction/data-frames.md). In this guide, you'll see how to migrate a simple example.
```